src/dataflow-types/src/client.rs

```diff
@@ -961,28 +1018,204 @@ pub mod process_local {
 
 /// A client to a remote dataflow server.
 pub mod tcp {
+    use mz_repr::proto::ProtoType;
+    use mz_repr::proto::RustType;
+    use mz_repr::proto::TryFromProtoError;
     use std::cmp;
     use std::fmt;
     use std::future::Future;
+    use std::net::ToSocketAddrs;
     use std::pin::Pin;
+    use std::sync::Arc;
     use std::time::Duration;
+    use tokio::sync::mpsc;
+    use tokio::sync::Mutex;
+    use tracing::info;
 
     use async_trait::async_trait;
     use futures::sink::SinkExt;
     use futures::stream::StreamExt;
+    use futures::Stream;
     use serde::de::DeserializeOwned;
     use serde::ser::Serialize;
     use tokio::io::{self, AsyncRead, AsyncWrite};
     use tokio::net::TcpStream;
+    use tokio::sync::mpsc::Receiver;
+    use tokio::sync::mpsc::Sender;
+    use tokio::sync::Notify;
     use tokio::time::{self, Instant};
     use tokio_serde::formats::Bincode;
+    use tokio_stream::wrappers::ReceiverStream;
     use tokio_util::codec::LengthDelimitedCodec;
+    use tonic::transport::Server;
+    use tonic::Request;
+    use tonic::Response;
+    use tonic::Streaming;
     use tracing::error;
 
+    use super::proto_compute_client::ProtoComputeClient;
+    use super::ComputeCommand;
+    use super::ComputeResponse;
+    use super::FromAddr;
+    use super::ProtoSerializableTimestamp;
     use crate::client::GenericClient;
 
     use super::Reconnect;
 
+    /// A client to a remote dataflow server using gRPC and protobuf based communication.
+    ///
+    /// The client opens a connection using the ProtoComputeClient stubs that
+    /// are generated by tonic from the service definition in `client.proto`. After creation, the client
+    /// is in disconnected state. To connect it, `connect` has to be called. Once the client
+    /// is connected, it will call automatically the only RPC defined in the service description: `Handle`.
+    /// Handle sets up two streams: A command stream going from this instance to the server and a response
+    /// stream coming back. Both of these streams are backed by one tokio mpsc queue each.
```

Comment:
is the one coming back from the server actually backed by an mpsc? it seems to just be the stream that the api `handle` call returns

Reply:
You're right